app/Filament/Resources/UserResource/Pages/Roles.php

```diff
@@ -123,4 +118,17 @@ public function table(Table $table): Table
                 Tables\Actions\BulkActionGroup::make([]),
             ]);
     }
+
+    private function removeDeveloperStaffRoles(User $targetUser): void
+    {
+        if ($targetUser->hasRole(Role::QUALITY_ASSURANCE)) {
+            $targetUser->removeRole(Role::QUALITY_ASSURANCE);
+        }
+        if ($targetUser->hasRole(Role::DEV_COMPLIANCE)) {
+            $targetUser->removeRole(Role::DEV_COMPLIANCE);
+        }
+        if ($targetUser->hasRole(Role::CODE_REVIEWER)) {
+            $targetUser->removeRole(Role::CODE_REVIEWER);
+        }
```

Comment:
These roles can be assigned to non-developers.

Reply:
As of latest, the staff dev roles can no longer be attached to a user unless they have the developer role first (may require a refresh when attaching the full dev role to a user).